pwnlib/util/packing.py

```diff
@@ -481,34 +484,41 @@ def make_unpacker(word_size = None, endianness = None, sign = None, **kwargs):
 
 
 
-def _flat(args, preprocessor, packer):
-    out = []
+def _flat(args, preprocessor, packer, address):
+    out = io.BytesIO()
     for arg in args:
+        address = address + out.tell()
+
+        while isinstance(arg, type(lambda:0)):
```

Comment:
Wouldn't it be better to check for `callable` here to support functions as well as lambdas?

Reply:
Should give precendence to `__flat_at__` / `__flat__` in that case though

Reply:
`callable` can't be used because we use `__call__` on `Label`

Reply:
It can be used if we check for `__flat__` / `__flat_at__` first, because Label has those.

Reply:
We can't check for `__flat__` etc. first, since we have to evaluate the callable / lambda *before* the check (or check again).

I added a special case check for `FitLabel` in 93584295